### DISCUSSION TITLE
127488 and 127489 Bit 1

### DISCUSSION THREAD

**Discussion by JSarv03** [canboat/canboat#447] [Q&A]
I believe this is a reference so the Network is aware of what engine is being monitored. Setting Bit 1 to (0) would indicate Port Engine or single engine? 

The reason I ask - I am working on getting a Deep Sea Electronics generator controller setup to communicate via J1939 Gateway to the NMEA Network - My last issue I am having - I have (2) engines I monitor through Fox Gateway for Mefi 3 ECM's - How can I set my transmitted data out so that the NMEA network knows this is not Port or Starboard Engine? I am worried this will cause a communication conflict as I am not sure that I can transmit that specific data (yet - I might be able to force that bit in the PLC of the control board) 

Any suggestions on how to ensure I will be able to monitor generator engine data without conflict of the port or starboard engine? I have access to PLC within the board that I can write bits to registers (which can set Canbus bits at specific addresses) I just haven't got that sorted out yet.

**Comment by sbender9** [comment]
Engine instance is a byte, so it can be greater than one.

**Reply by JSarv03** [reply]
So I would need to write "2" to byte 1 ?

**Reply by sbender9** [reply]
Yes

**Reply by sbender9** [reply]
Details here  https://canboat.github.io/canboat/canboat.html#pgn-list

**Reply by JSarv03** [reply]
Scott, thank you for taking the time to reply. I do have one last question. Since I likely cannot complete the handshake and comms directly over the N2k network I will need to run through a J1939 gateway. I have the ability to set any instrument value to any address I want. I assume I need to follow J1939 protocol to get my addressing correctly communicated through the gateway? 

Unfortunately I do not have enough information on the generator information to confirm it will work without modification. IE I don't know what is being shared automatically and I don't know what the gateway will communicate. I know the generator information is shared over a J1939-75 protocol and starts at address 44. The engine data will be standard J1939. 

Any insite into this? I almost need a gateway that I can set incoming information to the correct N2k network address.

**Reply by sbender9** [reply]
You definitely need a j1939 to n2k gateway. It will deal with the n2k side of things.

**Reply by sbender9** [reply]
I must admit I'm a little confused about what you are trying to do...

**Comment by JSarv03** [comment]
That makes two of us.

I have a 2001 340 DA with a Westerbeke 4.5 generator. I am going to install a Deep Sea Electronics generator controller to cover a few things for me. Control over the generator, remote monitoring, and additional sensors (for remote monitoring) 

Since I don't want to mount the control board in a generally easy to see location (eye sore) I would like to be able to monitor it  on my Garmin. 

I will be watching AC bus, Shore, and Generator. I'm fitting a CT on the bus side of the AC system to monitor usage as well. 

Engine sensors will be fitted to the generator for monitoring. 

Temperature sensors will be installed in the ER and Cabin. A water sensor (probably just a float) will be installed in the mid bilge and ER bilge. 

I will have a remote gateway mounted for real time monitoring and alarms. 

I do retrofits (retrofit design) on stationary generators as a small portion of my job. 

More of a novelty than anything else. 

My last issue will be if the J1939 gateway will register the 65004-65030 PGN's for AC data. I am not that it will....

**Comment by keesverruijt** [comment]
We don't know your J1939 gateway, or setup, but generally I believe that in the J1939 world addresses are fixed / hardwired / assigned, whereas I am sure that in NMEA 2000 they are automatically dynamically assigned. The NMEA 2000 address claim process includes a "name" of the device that ends up with an address; generally you do not work with addresses but names in end-user applications such as chart plotters.
You need to contact the engine or J1939 interface manufacturers and ask them whether it is even possible to have two engines on the same J1939 bus, and if so, ask the J1939/N2K gateway manufacturer whether the gateway will support this.

It may very well be that it is simpler to get two gateways, and set up two J1939 buses, one for each engine.

**Reply by JSarv03** [reply]
Well I am trying to determine what is "shared" from my controller. That seems to be the issue right now. I have 2 can options - ECM and CAN - I am not sure that any information is transmitted from the control board "automatically" IE its more of a read only Canbus device. However, I do have the option to Transmit Instrumentation but I must set the Message ID and all information associated with it. For example I pick "Engine Speed" as my source but I must set the Message ID, byte, bit, length, and conversion. 

Unfortunately these boards are designed around Modbus RTU/TCP. I have a register list and know how it works (it works well). But I cannot find enough information on Modbus to N2k gateways to make me feel comfortable enough to dive into to it knowing it will work without issues. 

I assume a Modbus RTU gateway would have a GUI of some type to allow me set registers to the respective locations needed on the N2k network. I can set my controller to slave or master - Master seems to work better than as a slave device. 

I know the Modbus Register Location and Data where engine speed, and I know where it needs to go - but I need to get from 1 protocol (RS485/Modbus RTU) to N2K.... 

Canbus may not be possible with this control board.
